java/memory/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -59,6 +62,21 @@ public void checkBuffers() {
   // ---------------------------------------- DEBUG ------------------------------------
 */
 
+  @Test
+  public void testMemoryOffset() {
+	try(final BufferAllocator allocator = new RootAllocator(10000)) {
+		int[] reqCapacities = {0, 5, 15, 510, 1024, 1023, 1025};
+        int[] expectedResult = {0, 8, 16, 512, 1024, 1024, 2048};
+        
+        for(int i = 0; i < reqCapacities.length; i++) {
+        	ArrowBuf buf = allocator.buffer(reqCapacities[i]);
+        	long memory = buf.memoryAddress();
+        	Assert.assertTrue("memory is align to 64-bytes", memory % CACHE_ALIGN == 0);
+        	Assert.assertEquals(expectedResult[i], buf.capacity());
+        	buf.release();
```

Comment:
we should check that offset + length does not go over underlying.capacity()
I'm curious to know the actual offsets on those buffers? Are they deterministic?
You have expectedResults on the capacities, could you have that for offsets?